src/data/equiv/algebra.lean

```diff
@@ -355,4 +380,112 @@ instance symm.is_ring_hom {e : α ≃r β} : is_ring_hom e.to_equiv.symm := hom
 @[simp] lemma to_equiv_symm_apply (e : α ≃r β) (x : β) :
   e.symm.to_equiv x = e.to_equiv.symm x := rfl
 
+/-- Two ring isomorphisms agree if they are defined by the
+    same underlying function. -/
+@[extensionality] lemma ext {R S : Type*} [ring R] [ring S]
```

Comment:
Are `ring_equiv`s defined for semirings? If so, please weaken the assumption to `semiring`.

Reply:
Done.